plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -964,4 +969,179 @@ private static boolean isNumericType(Type type)
+                        else {
+                            trueCount = OptionalLong.of(0);
+                        }
+                    }
+                    yield createBooleanColumnStatistics(
+                            falseCount,
+                            trueCount,
+                            nullsCount);
+                }
+                case BYTE, SHORT, INT, LONG -> createIntegerColumnStatistics(
+                        StringParser.toLong(columnStatistics.get(field + COLUMN_MIN)),
+                        StringParser.toLong(columnStatistics.get(field + COLUMN_MAX)),
+                        nullsCount,
+                        fromMetastoreDistinctValuesCount(distinctValuesCount, nullsCount, partitionRowCount));
+                case TIMESTAMP -> createIntegerColumnStatistics(
+                        OptionalLong.empty(),
+                        OptionalLong.empty(),
+                        nullsCount,
+                        fromMetastoreDistinctValuesCount(distinctValuesCount, nullsCount, partitionRowCount));
+                case FLOAT, DOUBLE -> createDoubleColumnStatistics(
+                        StringParser.toDouble(columnStatistics.get(field + COLUMN_MIN)),
+                        StringParser.toDouble(columnStatistics.get(field + COLUMN_MAX)),
+                        nullsCount,
+                        fromMetastoreDistinctValuesCount(distinctValuesCount, nullsCount, partitionRowCount));
+                case STRING, VARCHAR, CHAR -> createStringColumnStatistics(
+                        maxLength,
+                        getTotalSizeInBytes(avgLength, partitionRowCount, nullsCount),
+                        nullsCount,
+                        fromMetastoreDistinctValuesCount(distinctValuesCount, nullsCount, partitionRowCount));
+                case DATE -> createDateColumnStatistics(
+                        StringParser.toDate(columnStatistics.get(field + COLUMN_MIN)),
+                        StringParser.toDate(columnStatistics.get(field + COLUMN_MAX)),
+                        nullsCount,
+                        fromMetastoreDistinctValuesCount(distinctValuesCount, nullsCount, partitionRowCount));
+                case BINARY -> createBinaryColumnStatistics(
+                        maxLength,
+                        getTotalSizeInBytes(avgLength, partitionRowCount, nullsCount),
+                        nullsCount);
+                case DECIMAL -> createDecimalColumnStatistics(
+                        StringParser.toDecimal(columnStatistics.get(field + COLUMN_MIN)),
+                        StringParser.toDecimal(columnStatistics.get(field + COLUMN_MAX)),
+                        nullsCount,
+                        fromMetastoreDistinctValuesCount(distinctValuesCount, nullsCount, partitionRowCount));
+                case TIMESTAMPLOCALTZ, INTERVAL_YEAR_MONTH, INTERVAL_DAY_TIME, VOID, UNKNOWN -> null;
+            };
+        }
+    }
+
+    private static class StringParser
```

Comment:
This name is rather generic.
Please make it more specific - it is related to hive stats & spark